pkg/gateway/server/oauth.go

```diff
@@ -30,10 +30,7 @@ func (s *Server) oauth(apiContext api.Context) error {
 	}
 
 	// Check to make sure this auth provider exists.
-	list, err := s.dispatcher.ListConfiguredAuthProviders(apiContext.Context(), namespace)
-	if err != nil {
-		return fmt.Errorf("could not list configured auth providers: %w", err)
-	} else if !slices.Contains(list, name) {
+	if providerList := s.dispatcher.ListConfiguredAuthProviders(namespace); !slices.Contains(providerList, name) {
```

Comment:
For example, it would be find to continue to pass `apiContext.Context()` here.

Reply:
This is a different function. The `List` function just returns the stored list. The `Update` function is what refreshes the list.

Reply:
Sorry for my confusion here.

Reply:
No worries. I do think it is better to pass the context in like you suggested, so I made that change